authenticate/authenticate.go

```diff
@@ -191,3 +199,30 @@ func New(opts config.Options) (*Authenticate, error) {
 		templates: template.Must(frontend.NewTemplates()),
 	}, nil
 }
+
+// NewIdentityProvider returns a new identity provider based on its name.
+func NewIdentityProvider(o identity.Options) (a identity.Authenticator, err error) {
+	ctx := context.Background()
+	switch o.ProviderName {
+	case azure.Name:
+		a, err = azure.New(ctx, &o)
+	case gitlab.Name:
+		a, err = gitlab.New(ctx, &o)
+	case github.Name:
+		a, err = github.New(ctx, &o)
+	case google.Name:
+		a, err = google.New(ctx, &o)
+	case pom_oidc.Name:
+		a, err = pom_oidc.New(ctx, &o)
+	case okta.Name:
+		a, err = okta.New(ctx, &o)
+	case onelogin.Name:
+		a, err = onelogin.New(ctx, &o)
+	default:
+		return nil, fmt.Errorf("authenticate: unknown provider: %s", o.ProviderName)
+	}
+	if err != nil {
+		return nil, err
+	}
+	return a, nil
+}
```

Comment:
Just thoughts: not sure if it'll improve anything, would you be kind to help me know if there was any consideration for putting the `NewIdentityProvider ` function in the `authenticate` or in the `identity` package?